// @ariakit/react-core/cjs/__chunks/2MWKJ5UM.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _QXXNZT2Icjs = require('./QXXNZT2I.cjs');



var _GCJBB4ZTcjs = require('./GCJBB4ZT.cjs');


var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');

// src/combobox/combobox-context.tsx
var _react = require('react');
var ComboboxListRoleContext = _react.createContext.call(void 0, 
  void 0
);
var ctx = _WBFXWJUHcjs.createStoreContext.call(void 0, 
  [_QXXNZT2Icjs.PopoverContextProvider, _GCJBB4ZTcjs.CompositeContextProvider],
  [_QXXNZT2Icjs.PopoverScopedContextProvider, _GCJBB4ZTcjs.CompositeScopedContextProvider]
);
var useComboboxContext = ctx.useContext;
var useComboboxScopedContext = ctx.useScopedContext;
var useComboboxProviderContext = ctx.useProviderContext;
var ComboboxContextProvider = ctx.ContextProvider;
var ComboboxScopedContextProvider = ctx.ScopedContextProvider;
var ComboboxItemValueContext = _react.createContext.call(void 0, 
  void 0
);
var ComboboxItemCheckedContext = _react.createContext.call(void 0, false);










exports.ComboboxListRoleContext = ComboboxListRoleContext; exports.useComboboxContext = useComboboxContext; exports.useComboboxScopedContext = useComboboxScopedContext; exports.useComboboxProviderContext = useComboboxProviderContext; exports.ComboboxContextProvider = ComboboxContextProvider; exports.ComboboxScopedContextProvider = ComboboxScopedContextProvider; exports.ComboboxItemValueContext = ComboboxItemValueContext; exports.ComboboxItemCheckedContext = ComboboxItemCheckedContext;
