// @ariakit/react-core/cjs/__chunks/4FK7OQJV.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _ZTE44JRLcjs = require('./ZTE44JRL.cjs');


var _WILPQ3PLcjs = require('./WILPQ3PL.cjs');


var _64BKHVGOcjs = require('./64BKHVGO.cjs');




var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');


var _MZ2HG624cjs = require('./MZ2HG624.cjs');

// src/menubar/menubar.tsx
var _jsxruntime = require('react/jsx-runtime');
var TagName = "div";
var useMenubar = _WBFXWJUHcjs.createHook.call(void 0, 
  function useMenubar2({
    store: storeProp,
    composite = true,
    orientation: orientationProp,
    virtualFocus,
    focusLoop,
    rtl,
    ...props
  }) {
    const context = _ZTE44JRLcjs.useMenubarProviderContext.call(void 0, );
    storeProp = storeProp || context;
    const store = _WILPQ3PLcjs.useMenubarStore.call(void 0, {
      store: storeProp,
      orientation: orientationProp,
      virtualFocus,
      focusLoop,
      rtl
    });
    const orientation = store.useState(
      (state) => !composite || state.orientation === "both" ? void 0 : state.orientation
    );
    props = _MZ2HG624cjs.useWrapElement.call(void 0, 
      props,
      (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _ZTE44JRLcjs.MenubarScopedContextProvider, { value: store, children: element }),
      [store]
    );
    if (composite) {
      props = {
        role: "menubar",
        "aria-orientation": orientation,
        ...props
      };
    }
    props = _64BKHVGOcjs.useComposite.call(void 0, { store, composite, ...props });
    return props;
  }
);
var Menubar = _WBFXWJUHcjs.forwardRef.call(void 0, function Menubar2(props) {
  const htmlProps = useMenubar(props);
  return _WBFXWJUHcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useMenubar = useMenubar; exports.Menubar = Menubar;
