// @ariakit/react-core/cjs/__chunks/5WCU5NVK.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _57Z4CQLBcjs = require('./57Z4CQLB.cjs');


var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');

// src/dialog/dialog-context.tsx
var _react = require('react');
var ctx = _WBFXWJUHcjs.createStoreContext.call(void 0, 
  [_57Z4CQLBcjs.DisclosureContextProvider],
  [_57Z4CQLBcjs.DisclosureScopedContextProvider]
);
var useDialogContext = ctx.useContext;
var useDialogScopedContext = ctx.useScopedContext;
var useDialogProviderContext = ctx.useProviderContext;
var DialogContextProvider = ctx.ContextProvider;
var DialogScopedContextProvider = ctx.ScopedContextProvider;
var DialogHeadingContext = _react.createContext.call(void 0, void 0);
var DialogDescriptionContext = _react.createContext.call(void 0, void 0);









exports.useDialogContext = useDialogContext; exports.useDialogScopedContext = useDialogScopedContext; exports.useDialogProviderContext = useDialogProviderContext; exports.DialogContextProvider = DialogContextProvider; exports.DialogScopedContextProvider = DialogScopedContextProvider; exports.DialogHeadingContext = DialogHeadingContext; exports.DialogDescriptionContext = DialogDescriptionContext;
