// @ariakit/react-core/cjs/__chunks/6DC6MIBC.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _QXXNZT2Icjs = require('./QXXNZT2I.cjs');



var _GCJBB4ZTcjs = require('./GCJBB4ZT.cjs');


var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');

// src/select/select-context.tsx
var _react = require('react');
var ctx = _WBFXWJUHcjs.createStoreContext.call(void 0, 
  [_QXXNZT2Icjs.PopoverContextProvider, _GCJBB4ZTcjs.CompositeContextProvider],
  [_QXXNZT2Icjs.PopoverScopedContextProvider, _GCJBB4ZTcjs.CompositeScopedContextProvider]
);
var useSelectContext = ctx.useContext;
var useSelectScopedContext = ctx.useScopedContext;
var useSelectProviderContext = ctx.useProviderContext;
var SelectContextProvider = ctx.ContextProvider;
var SelectScopedContextProvider = ctx.ScopedContextProvider;
var SelectItemCheckedContext = _react.createContext.call(void 0, false);
var SelectHeadingContext = _react.createContext.call(void 0, null);









exports.useSelectContext = useSelectContext; exports.useSelectScopedContext = useSelectScopedContext; exports.useSelectProviderContext = useSelectProviderContext; exports.SelectContextProvider = SelectContextProvider; exports.SelectScopedContextProvider = SelectScopedContextProvider; exports.SelectItemCheckedContext = SelectItemCheckedContext; exports.SelectHeadingContext = SelectHeadingContext;
