// @ariakit/react-core/cjs/__chunks/AGBM3YM2.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _GCJBB4ZTcjs = require('./GCJBB4ZT.cjs');


var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');

// src/tag/tag-context.tsx
var _react = require('react');
var TagValueContext = _react.createContext.call(void 0, null);
var TagRemoveIdContext = _react.createContext.call(void 0, 
  null
);
var ctx = _WBFXWJUHcjs.createStoreContext.call(void 0, 
  [_GCJBB4ZTcjs.CompositeContextProvider],
  [_GCJBB4ZTcjs.CompositeScopedContextProvider]
);
var useTagContext = ctx.useContext;
var useTagScopedContext = ctx.useScopedContext;
var useTagProviderContext = ctx.useProviderContext;
var TagContextProvider = ctx.ContextProvider;
var TagScopedContextProvider = ctx.ScopedContextProvider;









exports.TagValueContext = TagValueContext; exports.TagRemoveIdContext = TagRemoveIdContext; exports.useTagContext = useTagContext; exports.useTagScopedContext = useTagScopedContext; exports.useTagProviderContext = useTagProviderContext; exports.TagContextProvider = TagContextProvider; exports.TagScopedContextProvider = TagScopedContextProvider;
