// @ariakit/react-core/cjs/__chunks/JIJ3G2UA.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});





var _ZTE44JRLcjs = require('./ZTE44JRL.cjs');



var _TPPZG3JZcjs = require('./TPPZG3JZ.cjs');



var _GCJBB4ZTcjs = require('./GCJBB4ZT.cjs');


var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');

// src/menu/menu-context.tsx
var _react = require('react');
var menu = _WBFXWJUHcjs.createStoreContext.call(void 0, 
  [_GCJBB4ZTcjs.CompositeContextProvider, _TPPZG3JZcjs.HovercardContextProvider],
  [_GCJBB4ZTcjs.CompositeScopedContextProvider, _TPPZG3JZcjs.HovercardScopedContextProvider]
);
var useMenuContext = menu.useContext;
var useMenuScopedContext = menu.useScopedContext;
var useMenuProviderContext = menu.useProviderContext;
var MenuContextProvider = menu.ContextProvider;
var MenuScopedContextProvider = menu.ScopedContextProvider;
var useMenuBarContext = _ZTE44JRLcjs.useMenubarContext;
var useMenuBarScopedContext = _ZTE44JRLcjs.useMenubarScopedContext;
var useMenuBarProviderContext = _ZTE44JRLcjs.useMenubarProviderContext;
var MenuBarContextProvider = _ZTE44JRLcjs.MenubarContextProvider;
var MenuBarScopedContextProvider = _ZTE44JRLcjs.MenubarScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);













exports.useMenuContext = useMenuContext; exports.useMenuScopedContext = useMenuScopedContext; exports.useMenuProviderContext = useMenuProviderContext; exports.MenuContextProvider = MenuContextProvider; exports.MenuScopedContextProvider = MenuScopedContextProvider; exports.useMenuBarContext = useMenuBarContext; exports.useMenuBarScopedContext = useMenuBarScopedContext; exports.useMenuBarProviderContext = useMenuBarProviderContext; exports.MenuBarContextProvider = MenuBarContextProvider; exports.MenuBarScopedContextProvider = MenuBarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
