// @ariakit/react-core/cjs/__chunks/WMHF7XQL.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});

var _5WCU5NVKcjs = require('./5WCU5NVK.cjs');


var _72TB42XGcjs = require('./72TB42XG.cjs');




var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');


var _MZ2HG624cjs = require('./MZ2HG624.cjs');

// src/dialog/dialog-dismiss.tsx
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
var TagName = "button";
var useDialogDismiss = _WBFXWJUHcjs.createHook.call(void 0, 
  function useDialogDismiss2({ store, ...props }) {
    const context = _5WCU5NVKcjs.useDialogScopedContext.call(void 0, );
    store = store || context;
    const onClickProp = props.onClick;
    const onClick = _MZ2HG624cjs.useEvent.call(void 0, (event) => {
      onClickProp == null ? void 0 : onClickProp(event);
      if (event.defaultPrevented) return;
      store == null ? void 0 : store.hide();
    });
    const children = _react.useMemo.call(void 0, 
      () => /* @__PURE__ */ _jsxruntime.jsxs.call(void 0, 
        "svg",
        {
          "aria-label": "Dismiss popup",
          display: "block",
          fill: "none",
          stroke: "currentColor",
          strokeLinecap: "round",
          strokeLinejoin: "round",
          strokeWidth: 1.5,
          viewBox: "0 0 16 16",
          height: "1em",
          width: "1em",
          children: [
            /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "line", { x1: "4", y1: "4", x2: "12", y2: "12" }),
            /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "line", { x1: "4", y1: "12", x2: "12", y2: "4" })
          ]
        }
      ),
      []
    );
    props = {
      "data-dialog-dismiss": "",
      children,
      ...props,
      onClick
    };
    props = _72TB42XGcjs.useButton.call(void 0, props);
    return props;
  }
);
var DialogDismiss = _WBFXWJUHcjs.forwardRef.call(void 0, function DialogDismiss2(props) {
  const htmlProps = useDialogDismiss(props);
  return _WBFXWJUHcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useDialogDismiss = useDialogDismiss; exports.DialogDismiss = DialogDismiss;
