// @ariakit/react-core/cjs/__chunks/ZTE44JRL.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});


var _GCJBB4ZTcjs = require('./GCJBB4ZT.cjs');


var _WBFXWJUHcjs = require('./WBFXWJUH.cjs');

// src/menubar/menubar-context.tsx
var _react = require('react');
var menubar = _WBFXWJUHcjs.createStoreContext.call(void 0, 
  [_GCJBB4ZTcjs.CompositeContextProvider],
  [_GCJBB4ZTcjs.CompositeScopedContextProvider]
);
var useMenubarContext = menubar.useContext;
var useMenubarScopedContext = menubar.useScopedContext;
var useMenubarProviderContext = menubar.useProviderContext;
var MenubarContextProvider = menubar.ContextProvider;
var MenubarScopedContextProvider = menubar.ScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);








exports.useMenubarContext = useMenubarContext; exports.useMenubarScopedContext = useMenubarScopedContext; exports.useMenubarProviderContext = useMenubarProviderContext; exports.MenubarContextProvider = MenubarContextProvider; exports.MenubarScopedContextProvider = MenubarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
