// @wordpress/components/build-module/validated-form-controls/components/input-control.js
// packages/components/src/validated-form-controls/components/input-control.tsx
import { forwardRef, useRef } from "@wordpress/element";
import { useMergeRefs } from "@wordpress/compose";
import { ControlWithError } from "../control-with-error";
import InputControl from "../../input-control";
import { jsx as _jsx } from "react/jsx-runtime";
var UnforwardedValidatedInputControl = ({
  required,
  onValidate,
  customValidity,
  onChange,
  markWhenOptional,
  ...restProps
}, forwardedRef) => {
  const validityTargetRef = useRef(null);
  const mergedRefs = useMergeRefs([forwardedRef, validityTargetRef]);
  const valueRef = useRef(restProps.value);
  return /* @__PURE__ */ _jsx(ControlWithError, {
    required,
    markWhenOptional,
    onValidate: () => {
      return onValidate?.(valueRef.current);
    },
    customValidity,
    getValidityTarget: () => validityTargetRef.current,
    children: /* @__PURE__ */ _jsx(InputControl, {
      __next40pxDefaultSize: true,
      ref: mergedRefs,
      onChange: (value, ...args) => {
        valueRef.current = value;
        onChange?.(value, ...args);
      },
      ...restProps
    })
  });
};
var ValidatedInputControl = forwardRef(UnforwardedValidatedInputControl);
export {
  ValidatedInputControl
};
